documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
 
-==== Building the kernel
+TIP: If you don't want to install the freshly-compiled kernel onto the Raspberry Pi where you run this command, copy the compiled kernel to the boot partition of a separate boot media instead of `/boot/firmware/`.
 
-Build and install the kernel, modules, and Device Tree blobs. This step can take a long time depending on the Raspberry Pi model you are using.  For the 32-bit kernel:
+To build and install the 64-bit kernel, run the following commands:
 
-[,bash]
+[source,console]
 ----
-make -j4 zImage modules dtbs
-sudo make modules_install
-# Choose one of the following based on the kernel version
-  # For kernels up to 6.4:
-  sudo cp arch/arm/boot/dts/*.dtb /boot/firmware/
-  # For kernel 6.5 and above:
-  sudo cp arch/arm/boot/dts/broadcom/*.dtb /boot/firmware/
-sudo cp arch/arm/boot/dts/overlays/*.dtb* /boot/firmware/overlays/
-sudo cp arch/arm/boot/dts/overlays/README /boot/firmware/overlays/
-sudo cp arch/arm/boot/zImage /boot/firmware/$KERNEL.img
+$ make -j4 Image.gz modules dtbs
+$ sudo make modules_install
+$ sudo cp arch/arm64/boot/dts/broadcom/*.dtb /boot/firmware/
+$ sudo cp arch/arm64/boot/dts/overlays/*.dtb* /boot/firmware/overlays/
+$ sudo cp arch/arm64/boot/dts/overlays/README /boot/firmware/overlays/
+$ sudo cp arch/arm64/boot/Image.gz /boot/firmware/$KERNEL.img
 ----
 
-For the 64-bit kernel:
+To build the 32-bit kernel:
 
-[,bash]
+. Run the following commands to build the modules and Device tree blobs:
++
+[source,console]
 ----
-make -j4 Image.gz modules dtbs
-sudo make modules_install
-sudo cp arch/arm64/boot/dts/broadcom/*.dtb /boot/firmware/
-sudo cp arch/arm64/boot/dts/overlays/*.dtb* /boot/firmware/overlays/
-sudo cp arch/arm64/boot/dts/overlays/README /boot/firmware/overlays/
-sudo cp arch/arm64/boot/Image.gz /boot/firmware/$KERNEL.img
+$ make -j4 zImage modules dtbs
+$ sudo make modules_install
 ----
 
-NOTE: On Raspberry Pi 2, 3, 4 and 5, the `-j4` flag splits the work between all four cores, speeding up compilation significantly.
+. Depending on your kernel version, run the following command:
+  * For kernels up to version 6.4:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/*.dtb /boot/firmware/
+----
+* For kernels version 6.5 and above:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/broadcom/*.dtb /boot/firmware/
+----
+. Finally, copy over the overlays, README, and image:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/overlays/*.dtb* /boot/firmware/overlays/
+$ sudo cp arch/arm/boot/dts/overlays/README /boot/firmware/overlays/
+$ sudo cp arch/arm/boot/zImage /boot/firmware/$KERNEL.img
+----
 
-If you now reboot, your Raspberry Pi should be running your freshly-compiled kernel.
+TIP: On multi-core Raspberry Pi models, the `-j <n>` option distributes work between cores. This can speed up compilation significantly. Run `nproc` to see how many processors you have; we recommend passing a number 1.5x your number of processors.
```

Comment:
Based on that recommendation, should the example commands above actually be using `-j 6` rather than `-j 4` ? :wink: 
Also, perhaps "...the `-j <n>` option..." wording should be expanded here slightly to "...the `make -j <n>` option...", to make it more obvious what it's referring to?